### PR TITLE
Mapping and generating Default CCJ - Mapping 

### PR DESCRIPTION
**Pull request by karthikkannankainos**
Change description

1. CCD domain change for CCJ
2. Mapper changes.
3. Test changes.

https://tools.hmcts.net/jira/browse/ROC-4754

### CODE REVIEW

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/domain/CCDStatementOfTruth.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package uk.gov.hmcts.cmc.ccd.domain;
+
+import lombok.Builder;
+import lombok.Value;
+
+@Value
+@Builder
+public class CCDStatementOfTruth {
```
Comment (sabahirfan) [inline ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/domain/CCDStatementOfTruth.java]: This is not required with new definition.

File: ccd-adapter/src/test/java/uk/gov/hmcts/cmc/ccd/mapper/ccj/CountyCourtJudgmentMapperTest.java [hunk 2]
```diff
@@ -63,9 +67,11 @@ public void shouldMapCountyCourtJudgmentWithPaymentOptionImmediatelyToCCD() {
     public void shouldMapCountyCourtJudgmentWithPaymentOptionSpecifiedDateToCCD() {
         //given
         final CountyCourtJudgment countyCourtJudgment = SampleCountyCourtJudgment.builder()
+            .ccjType(CountyCourtJudgmentType.ADMISSIONS)
             .paymentOption(PaymentOption.BY_SPECIFIED_DATE)
             .payBySetDate(now())
             .statementOfTruth(new StatementOfTruth("Tester", "Unit Test"))
+            .ccjType(CountyCourtJudgmentType.DEFAULT)
```
Comment (sabahirfan) [inline ccd-adapter/src/test/java/uk/gov/hmcts/cmc/ccd/mapper/ccj/CountyCourtJudgmentMapperTest.java]: Either need this line or Line 70, but not both.

Comment (karthikkannankainos) [reply ccd-adapter/src/test/java/uk/gov/hmcts/cmc/ccd/mapper/ccj/CountyCourtJudgmentMapperTest.java]: removed

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/domain/ccj/CCDCountyCourtJudgment.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    private CCDPaymentOption paymentOption;
+    private BigDecimal repaymentPlanInstalmentAmount;
+    private LocalDate repaymentPlanFirstPaymentDate;
+    private LocalDate repaymentPlanCompletionDate;
+    private String repaymentPlanPaymentLength;
+    private CCDPaymentSchedule repaymentPlanPaymentSchedule;
+    private LocalDate payBySetDate;
+    private String statementOfTruthSignerName;
+    private String statementOfTruthSignerRole;
+    private CCDCountyCourtJudgmentType ccjType;
```
Comment (ckalista) [inline ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/domain/ccj/CCDCountyCourtJudgment.java]: This needs changing to `type` as agreed during our latest conversation but let's do it in a separate PR.

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/ccj/CountyCourtJudgmentMapper.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    @Override
+    public CCDCountyCourtJudgment to(CountyCourtJudgment countyCourtJudgment) {
+
+        CCDCountyCourtJudgment.CCDCountyCourtJudgmentBuilder builder = CCDCountyCourtJudgment.builder();
+
+        countyCourtJudgment.getDefendantDateOfBirth().ifPresent(builder::defendantDateOfBirth);
+        countyCourtJudgment.getPaidAmount().ifPresent(builder::paidAmount);
+        builder.paymentOption(valueOf(countyCourtJudgment.getPaymentOption().name()));
+        countyCourtJudgment.getRepaymentPlan().ifPresent(repaymentPlan -> {
+            builder.repaymentPlanFirstPaymentDate(repaymentPlan.getFirstPaymentDate());
```
[the remark this answers is not in the document]
Comment (karthikkannankainos) [reply ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/ccj/CountyCourtJudgmentMapper.java]: rp was removed and repaymentPlan was added

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/ccj/CountyCourtJudgmentMapper.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    }
+
+    @Override
+    public CountyCourtJudgment from(CCDCountyCourtJudgment ccdCountyCourtJudgment) {
+
+        return new CountyCourtJudgment(
+            ccdCountyCourtJudgment.getDefendantDateOfBirth(),
+            PaymentOption.valueOf(ccdCountyCourtJudgment.getPaymentOption().name()),
+            ccdCountyCourtJudgment.getPaidAmount(),
+            RepaymentPlan.builder().paymentLength(ccdCountyCourtJudgment.getRepaymentPlanPaymentLength())
```
[the remark this answers is not in the document]
Comment (karthikkannankainos) [reply ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/ccj/CountyCourtJudgmentMapper.java]: fixed it

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/ccj/CountyCourtJudgmentMapper.java [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+            builder.ccjType(CCDCountyCourtJudgmentType.valueOf(ccjType.name()))
+        );
+
+        return builder.build();
+    }
+
+    @Override
+    public CountyCourtJudgment from(CCDCountyCourtJudgment ccdCountyCourtJudgment) {
+
+        return new CountyCourtJudgment(
```
[the remark this answers is not in the document]
Comment (karthikkannankainos) [reply ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/ccj/CountyCourtJudgmentMapper.java]: the previous code used to have @Builder in the constructor and used the constructor to build the same. But now I have changed to builder.

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/ccj/CountyCourtJudgmentMapper.java [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+            PaymentOption.valueOf(ccdCountyCourtJudgment.getPaymentOption().name()),
+            ccdCountyCourtJudgment.getPaidAmount(),
+            RepaymentPlan.builder().paymentLength(ccdCountyCourtJudgment.getRepaymentPlanPaymentLength())
+                .instalmentAmount(ccdCountyCourtJudgment.getRepaymentPlanInstalmentAmount())
+                .firstPaymentDate(ccdCountyCourtJudgment.getRepaymentPlanFirstPaymentDate())
+                .completionDate(ccdCountyCourtJudgment.getRepaymentPlanCompletionDate())
+                .paymentSchedule(
+                    PaymentSchedule.valueOf(ccdCountyCourtJudgment.getRepaymentPlanPaymentSchedule().name()))
+                .build(),
+            ccdCountyCourtJudgment.getPayBySetDate(),
```
[the remark this answers is not in the document]
Comment (karthikkannankainos) [reply ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/ccj/CountyCourtJudgmentMapper.java]: but still it will be null while creating the object. Repayment plan will be now built only if the value from CCJ object is available. Added extra null check

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/Mapper.java [hunk 8]
```diff
@@ -1,4 +1,4 @@
-package uk.gov.hmcts.cmc.ccd.domain.mapper;
+package uk.gov.hmcts.cmc.ccd.mapper;
```
[the remark this answers is not in the document]
Comment (karthikkannankainos) [reply ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/Mapper.java]: domain and mapper are at same level in deprecated too. I feel it can be under ccd package as exclusive ones.

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/ccj/CountyCourtJudgmentMapper.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (srineelesh) [reply ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/ccj/CountyCourtJudgmentMapper.java]: no .. rp has been removed but that is in a different place that is inside paymentIntention

[the remark this answers is not in the document]
Comment (karthikkannankainos) [reply ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/ccj/CountyCourtJudgmentMapper.java]: repaymentPlan  prefix still stays in Sanchita mapping. if its changed then it need to go as separate PR
